src/Widgets/TaskFormRevealer.vala

```diff
@@ -0,0 +1,173 @@
+/*
+* Copyright 2019 elementary, Inc. (https://elementary.io)
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public
+* License as published by the Free Software Foundation; either
+* version 3 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have recei ved a copy of the GNU General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*
+*/
+
+public class Tasks.TaskFormRevealer : Gtk.Revealer {
+
+    public ECal.Component task { get; construct set; }
+
+    private Gtk.Switch due_switch;
+    private Gtk.TextBuffer description_textbuffer;
+    private Granite.Widgets.DatePicker due_datepicker;
+    private Granite.Widgets.TimePicker due_timepicker;
+
+    public signal void cancel_clicked ();
+    public signal void save_clicked (ECal.Component task);
+    public signal void delete_clicked (ECal.Component task);
+
+    public TaskFormRevealer (ECal.Component task) {
+        Object (task: task);
+    }
+
+    construct {
+        due_switch = new Gtk.Switch ();
+        due_switch.valign = Gtk.Align.CENTER;
+
+        var due_label = new Gtk.Label (_("Schedule:"));
+        due_datepicker = new Granite.Widgets.DatePicker ();
+        due_timepicker = new Granite.Widgets.TimePicker ();
+
+        var due_datetimepicker = new Gtk.Box (Gtk.Orientation.HORIZONTAL, 12);
+        due_datetimepicker.add (due_label);
+        due_datetimepicker.add (due_switch);
+        due_datetimepicker.add (due_datepicker);
+        due_datetimepicker.add (due_timepicker);
+
+        var description_textview = new Gtk.TextView ();
+        description_textview.margin = 12;
```

Comment:
You probably want `border_width` instead of `margin` since `border_width` is internal to the text view and you won't get that grey around the outside

Reply:
Fixed.